flink-runtime/src/main/java/org/apache/flink/runtime/highavailability/JobResultEntry.java

```diff
@@ -35,4 +41,20 @@ public JobResultEntry(JobResult jobResult) {
     public JobResult getJobResult() {
         return jobResult;
     }
+
+    /** Builder for {@link JobResultEntry}. */
+    @Internal
+    public static class Builder {
```

Comment:
Is the `Builder` necessary for the Json serialization? Otherwise, it's not adding much value since we don't have a default value for `JobResult` provided. Essentially, the Builder does the same like the constructor. WDYT?

Thinking about it, we might want to add a `NonNull` check to `JobResult` since it's more of a required value. I will add this in the downstream branch.